### PR TITLE
Replace `GetFieldAccess` with indexing function in `SqlToRel `

### PR DESCRIPTION
**Pull request by jayzhan211**
## Which issue does this PR close?

Part of #10374

### CODE REVIEW

File: datafusion/sqllogictest/test_files/expr.slt [hunk 1]
```diff
@@ -2325,28 +2325,142 @@ host3 3.3
...
+        from t1
+    ) t2 
+    where server_host IS NOT NULL 
+    group by server_host order by host;
 ----
 host1 101
 host2 202
 host3 303
 
+# TODO: Issue tracked in https://github.com/apache/datafusion/issues/10364
```
Comment (jayzhan211) [inline datafusion/sqllogictest/test_files/expr.slt:2348]: This is the issue from #10364, not a regression caused by this change.

**Comment by alamb** [comment]
cc @westonpace

**Review by alamb — APPROVED** [review]
Looks good to me -- thank you @jayzhan211  -- I am going to also tag this PR as api change so it gets highlighted in the readme

**Comment by jayzhan211** [comment]
Thanks @alamb !

**Pull request merged**